sharktank/tests/evaluate/perplexity_test.py

```diff
@@ -6,344 +6,244 @@
 
-        llama_8b_perplexity = perplexity.main(
+        current_perplexity = perplexity.main(
             [
-                f"--gguf-file={llama_8b_fp8_gguf_path}",
-                f"--tokenizer-config-json={llama_8b_fp8_tokenizer_path}",
+                f"--gguf-file={self.llama3_8b_fp8_model}",
+                f"--tokenizer-config-json={self.llama3_8b_tokenizer}",
                 f"--attention-kernel=torch_sdpa",
             ]
         )
 
-        # dummy data
-        baseline_llama_8b_perplexity = {
-            "perplexities": [
-                9.875290870666504,
-                8.075149536132812,
-                16.164775848388672,
-                11.06580924987793,
-                11.46964168548584,
-                12.714613914489746,
-            ],
-            "mean_perplexity": 11.560880184173584,
-        }
-
-        delta = 5e-1
+        self.save_perplexity(model_name, current_perplexity)
 
         self.assertAlmostEqual(
-            baseline_llama_8b_perplexity["mean_perplexity"],
-            llama_8b_perplexity["mean_perplexity"],
-            delta=delta,
-            msg=f"Perplexity is deviating more than {delta}",
+            baseline_perplexity["mean_perplexity"],
+            current_perplexity["mean_perplexity"],
+            delta=self.delta,
+            msg=f"Perplexity is deviating more than {self.delta}",
         )
 
     @longrun
     def test_llama3_405B_f16_decomposed(self):
 
         # Llama 3.1 405B decomposed
 
-        llama_405b_f16_gguf_path = (
-            "/data/extra/models/llama3.1_405B/llama405b_fp16.gguf"
-        )
-        llama_405b_f16_tokenizer_path = (
-            "/data/extra/models/llama3.1_405B/tokenizer_config.json"
-        )
-
-        tensor_parallelism_size = 8
+        model_name = "llama3_405B_f16_decomposed"
+        baseline_perplexity = self.baseline_perplexity[model_name]
 
-        llama_405b_perplexity = perplexity.main(
+        current_perplexity = perplexity.main(
             [
-                f"--gguf-file={llama_405b_f16_gguf_path}",
-                f"--tokenizer-config-json={llama_405b_f16_tokenizer_path}",
-                f"--tensor-parallelism-size={tensor_parallelism_size}",
+                f"--gguf-file={self.llama3_405b_f16_model}",
+                f"--tokenizer-config-json={self.llama3_405b_tokenizer}",
+                f"--tensor-parallelism-size={self.tensor_parallelism_size}",
             ]
         )
 
-        baseline_llama_405b_perplexity = {
-            "perplexities": [
-                2.0203986167907715,
-                4.045348644256592,
-                4.452215671539307,
-                4.009974479675293,
-                5.169974327087402,
-                5.516016960144043,
-            ],
-            "mean_perplexity": 4.202321449915568,
-        }
-
-        delta = 5e-1
+        self.save_perplexity(model_name, current_perplexity)
 
         self.assertAlmostEqual(
```

Comment:
Should we be failing if current perplexity decreases and no longer matches baseline?

Reply:
It will fail if current perplexity increases or decreases to baseline by 0.5, so we can track what changed.